packages/bff/src/routers/clientRouter.ts

```diff
@@ -6,48 +6,286 @@ import {
+        const errorRes = makeApiProblem(error, emptyErrorMapper, ctx.logger);
+        return res.status(errorRes.status).json(errorRes).end();
+      }
+    })
+
+    .get("/clients/:clientId", async (req, res) => {
+      const ctx = fromBffAppContext(req.ctx, req.headers);
+      try {
+        const client = await clientService.getClientById(
+          req.params.clientId,
+          ctx
+        );
+
+        return res.status(200).json(client).end();
+      } catch (error) {
+        const errorRes = makeApiProblem(error, emptyErrorMapper, ctx.logger);
+        return res.status(errorRes.status).json(errorRes).end();
+      }
+    })
+
+    .delete("/clients/:clientId", async (req, res) => {
+      const ctx = fromBffAppContext(req.ctx, req.headers);
+      try {
+        await clientService.deleteClient(req.params.clientId, ctx);
+
+        return res.status(204).send();
+      } catch (error) {
+        const errorRes = makeApiProblem(error, emptyErrorMapper, ctx.logger);
+        return res.status(errorRes.status).json(errorRes).end();
+      }
+    })
+
+    .delete("/clients/:clientId/purposes/:purposeId", async (req, res) => {
+      const ctx = fromBffAppContext(req.ctx, req.headers);
+      try {
+        await clientService.removeClientPurpose(
+          req.params.clientId,
+          req.params.purposeId,
+          ctx
+        );
+
+        return res.status(204).send();
+      } catch (error) {
+        const errorRes = makeApiProblem(error, emptyErrorMapper, ctx.logger);
+        return res.status(errorRes.status).json(errorRes).end();
+      }
+    })
+
+    .get("/clients/:clientId/keys/:keyId", async (req, res) => {
+      const ctx = fromBffAppContext(req.ctx, req.headers);
+      try {
+        const key = await clientService.getClientKeyById(
+          req.params.clientId,
+          req.params.keyId,
+          ctx.authData.selfcareId,
+          ctx
+        );
+
+        return res.status(200).json(key).end();
+      } catch (error) {
+        const errorRes = makeApiProblem(error, emptyErrorMapper, ctx.logger);
+        return res.status(errorRes.status).json(errorRes).end();
+      }
+    })
+    .delete("/clients/:clientId/keys/:keyId", async (req, res) => {
+      const ctx = fromBffAppContext(req.ctx, req.headers);
+
+      try {
+        await clientService.deleteClientKeyById(
+          req.params.clientId,
+          req.params.keyId,
+          ctx
+        );
+
+        return res.status(204).send();
+      } catch (error) {
+        const errorRes = makeApiProblem(error, emptyErrorMapper, ctx.logger);
+        return res.status(errorRes.status).json(errorRes).end();
+      }
+    })
+
+    .post("/clients/:clientId/users/:userId", async (req, res) => {
+      const ctx = fromBffAppContext(req.ctx, req.headers);
+
+      try {
+        await clientService.addUserToClient(
+          req.params.userId,
+          req.params.clientId,
+          ctx
+        );
+
+        return res.status(204).send();
```

Comment:
According to the spec it seems that this should return 200 and a `CreatedResource`, not a 204 No Content. Could you double check? In that case, weird that TS does not complain here 🤔

Reply:
Yes in fact i should return a `CreatedResource`, apparently TS does not complain if you don't use `.json()` in the reply builder 🙃